torchmetrics/image/uqi.py

```diff
@@ -0,0 +1,98 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, List, Optional, Sequence
+
+import torch
+from torch import Tensor
+
+from torchmetrics.functional.image.uqi import _uqi_compute, _uqi_update
+from torchmetrics.metric import Metric
+from torchmetrics.utilities import _future_warning, rank_zero_warn
+from torchmetrics.utilities.data import dim_zero_cat
+
+
+class UniversalImageQualityIndex(Metric):
+    """Computes Universal Image Quality Index (UQI_).
+
+    Args:
+        kernel_size: size of the gaussian kernel
+        sigma: Standard deviation of the gaussian kernel
+        reduction: a method to reduce metric score over labels.
+
+            - ``'elementwise_mean'``: takes the mean (default)
+            - ``'sum'``: takes the sum
+            - ``'none'``: no reduction will be applied
+
+        data_range: Range of the image. If ``None``, it is determined from the image (max - min)
+
+    Return:
+        Tensor with UQI score
+
+    Example:
+        >>> from torchmetrics import UniversalImageQualityIndex
+        >>> preds = torch.rand([16, 1, 16, 16])
+        >>> target = preds * 0.75
+        >>> uqi = UniversalImageQualityIndex()
+        >>> uqi(preds, target)
+        tensor(0.9216)
+    """
+
+    preds: List[Tensor]
+    target: List[Tensor]
+    higher_is_better = True
+
+    def __init__(
+        self,
+        kernel_size: Sequence[int] = (11, 11),
+        sigma: Sequence[float] = (1.5, 1.5),
+        reduction: str = "elementwise_mean",
```

Comment:
```suggestion
        reduction: Literal["elementwise_mean", "sum", "none"] = "elementwise_mean",
```

see comments from @stancld